File: python/find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        A_set = set()
        B_set = set()
        C = []
        common = 0
        for i in range(len(A)):
            A_set.add(A[i])
            B_set.add(B[i])
            if A[i] in B_set:
                common += 1
            if B[i] in A_set:
                common += 1
            if A[i] == B[i]:
                common -= 1
            C.append(common)
        return C

File: go/find-the-prefix-common-array-of-two-arrays.go
func findThePrefixCommonArray(A []int, B []int) []int {
    A_set := make([]bool, len(A))
    B_set := make([]bool, len(A))
    C := make([]int, len(A))
    common := 0
    for i := 0; i < len(A); i++ {
        a := A[i] - 1
        b := B[i] - 1
        A_set[a] = true
        B_set[b] = true
        if B_set[a] {
            common += 1
        }
        if A_set[b] {
            common += 1
        }
        if a == b {
            common -= 1
        }
        C[i] = common
    }
    return C
}
